### PR TITLE
[fix] `jsx-uses-vars` ignores lowercase tag names

### PR DESCRIPTION
**Pull request by alanorozco**
Context: eslint/eslint#15040

### CODE REVIEW

File: lib/rules/jsx-uses-vars.js [hunk 1]
```diff
@@ -33,6 +36,10 @@ module.exports = {
         if (node.name.name) {
           // <Foo>
           name = node.name.name;
+          // Exclude lowercase tag names like <div>
+          if (isTagName(name)) {
```
Comment (ljharb) [inline lib/rules/jsx-uses-vars.js:40]: what does this do with namespaced tags? Is `<a.X />` not allowed?

Comment (alanorozco) [reply lib/rules/jsx-uses-vars.js:40]: `a.X` is not handled on this codepath, I believe that's under the condition below for `node.name.object`. [They're interpreted as references](https://babeljs.io/repl#?browsers=ie%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=DwPgUABBwGYPZwgenFYAxBzXXnAdAEICGATtpLgvgEZkXApA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react&prettier=true&targets=&version=7.15.5&externalPlugins=%40babel%2Fplugin-syntax-object-rest-spread%407.8.3&assumptions=%7B%7D) so they should work fine.

If you would like, I could add the following valid test cases.

```js
    {
      code: `
        /* eslint jsx-uses-vars: 1 */
        var object;
        React.render(<object.Tag />);
      `
    }, {
      code: `
        /* eslint jsx-uses-vars: 1 */
        var object;
        React.render(<object.tag />);
      `
    }
```

Comment (ljharb) [reply lib/rules/jsx-uses-vars.js:40]: Thanks, adding those would be great :-)

**Review by ljharb — APPROVED** [review]

**Comment by yoyo837** [comment]
Nice fix.

**Comment by mrtnzlml** [comment]
Hola @alanorozco! 👋 It seems like this PR caused a breaking change in a patch release, see: https://github.com/yannickcr/eslint-plugin-react/issues/3080

Could you please have a look? I am not sure what would be the right thing to do in this case. 🤔

**Pull request merged**